docs/05-plugin-development.md

```diff
@@ -63,16 +63,177 @@ Type: `string`
 
 The name of the plugin, for use in error messages and warnings.
 
-### Hooks
+### Build Hooks
 
-In addition to properties defining the identity of your plugin, you may also specify properties that correspond to available build hooks. Hooks can affect how a build is run, provide information about a build, or modify a build once complete. There are different kinds of hooks:
+To make your plugin functional, you add function properties that correspond to some of the various available hooks. They will be called by Rollup at specific points in the build process. Hooks can affect how a build is run, provide information about a build, or modify a build once complete. There are different kinds of hooks:
 
-* `async`: The hook can also return a promise resolving to the same type of value; otherwise, the hook is marked as `sync`
-* `first`: If several plugins implement this hook, the hooks are run sequentially until a hook returns a value other than `null` or `undefined`
-* `sequential`: If this hook returns a promise, then other hooks of this kind will only be executed once this hook has resolved
-* `parallel`: If this hook returns a promise, then other hooks of this kind will not wait for this hook to be resolved
+* `async`: The hook may also return a promise resolving to the same type of value; otherwise, the hook is marked as `sync`.
+* `first`: If several plugins implement this hook, the hooks are run sequentially until a hook returns a value other than `null` or `undefined`.
+* `sequential`: If several plugins implement this hook, all of them will be run in the specified plugin order. If a hook is async, subsequent hooks of this kind will wait until the current hook is resolved.
+* `parallel`: If several plugins implement this hook, all of them will be run in the specified plugin order. If a hook is async, subsequent hooks of this kind will be run in parallel and not wait for the current hook.
 
-Furthermore, hooks can be run either during the `build` phase of the Rollup build, which is triggered by `rollup.rollup()`, or during the `generate` phase, which is triggered by `bundle.generate()`  or `bundle.write()`. Plugins that only use `generate` phase hooks can also be passed in via the output options to `bundle.generate()` or `bundle.write()` and therefore run only for certain outputs.
+Build hooks are run during the build phase, which is triggered by `rollup.rollup(inputOptions)`. They are mainly concerned with locating, providing and transforming input files before they are processed by Rollup. The first hook of the build phase is [options](guide/en/#options), the last one is always [buildEnd](guide/en/#buidend). Additionally, the [watchChange](guide/en/#watchchange) hook can be triggered at any time.
```

Comment:
Yes, at the moment there is no mechanism to prevent this from happening. A rebuild, however, will only start once the current build is complete and sufficient time, at the moment 200ms, has passed without another change event. Maybe I should spell this out in more detail.